instrumentation/opentelemetry-instrumentation-mysqlclient/tests/test_mysqlclient_integration.py

```diff
@@ -96,6 +96,264 @@ def test_instrument_connection(self, mock_connect):
         spans_list = self.memory_exporter.get_finished_spans()
         self.assertEqual(len(spans_list), 1)
 
+    @mock.patch("opentelemetry.instrumentation.dbapi.instrument_connection")
+    @mock.patch("MySQLdb.connect")
+    # pylint: disable=unused-argument
+    def test_instrument_connection_enable_commenter_dbapi_kwargs(
+        self,
+        mock_connect,
+        mock_instrument_connection,
+    ):
+        cnx = MySQLdb.connect(database="test")
+        cnx = MySQLClientInstrumentor().instrument_connection(
+            cnx,
+            enable_commenter=True,
+            commenter_options={"foo": True},
+        )
+        cursor = cnx.cursor()
+        cursor.execute("Select 1;")
+        kwargs = mock_instrument_connection.call_args[1]
+        self.assertEqual(kwargs["enable_commenter"], True)
+        self.assertEqual(kwargs["commenter_options"], {"foo": True})
+
+    def test_instrument_connection_with_dbapi_sqlcomment_enabled(self):
+        mock_cursor = mock.MagicMock()
+        mock_cursor.execute = mock.MagicMock()
+        mock_connection = mock.MagicMock()
+        mock_connection.cursor.return_value = mock_cursor
+
+        mock_connect_module = mock.MagicMock()
+        mock_connect_module.__name__ = "MySQLdb"
+        mock_connect_module.threadsafety = "123"
+        mock_connect_module.apilevel = "123"
+        mock_connect_module.paramstyle = "test"
+        mock_connect_module._mysql.get_client_info = mock.Mock(
+            return_value="foobaz"
+        )
+        mock_connect_module.connect = mock.Mock(return_value=mock_connection)
+
+        with mock.patch(
+            "opentelemetry.instrumentation.mysqlclient.MySQLdb",
+            mock_connect_module,
+        ), mock.patch(
+            "opentelemetry.instrumentation.dbapi.util_version",
+            return_value="foobar",
+        ):
+            cnx_proxy = MySQLClientInstrumentor().instrument_connection(
+                mock_connection,
+                enable_commenter=True,
+            )
+            cnx_proxy.cursor().execute("Select 1;")
+            self.assertRegex(
+                mock_cursor.execute.call_args[0][0],
+                r"Select 1 /\*db_driver='MySQLdb%%3Afoobar',dbapi_level='123',dbapi_threadsafety='123',driver_paramstyle='test',mysql_client_version='foobaz',traceparent='\d{1,2}-[a-zA-Z0-9_]{32}-[a-zA-Z0-9_]{16}-\d{1,2}'\*/;",
```

Comment:
consider setting the trace context in the test so that the test is deterministic. That way you can avoid the regex since the string should be the same every time

Reply:
Changed to `assertEqual` using the resulting in-memory spans in [e0ecef0](https://github.com/open-telemetry/opentelemetry-python-contrib/pull/2941/commits/e0ecef01b9babd5e9e6ae6c07d09c92360b33e5d)